File: 5.2-integration/1.1-function-notebook/py/input.py
from util import validate_float, validate_range

def input_option(msg:str, start:int, end:int, input_msg:str="Enter Option: ") -> int:
    print(msg)
    option = input(input_msg)

    while(not validate_range(start=start, end=end, i=option)):
        print(f"Invalid Input! Range must be between {start} to {end}")
        option = input("Input : ")

    return int(option)

def input_float(msg:str="Enter a number : ") -> int:
    value = input(msg)

    while(not validate_float(value)):
        print(f"Invalid Input! It must be a number")
        value = input(msg)

    return float(value)

File: 5.2-integration/1.1-function-notebook/cpp/input.cpp
#include "input.h"

int input_option(int start, int end, std::string msg, std::string input_msg="Choose Option: "){
    std::cout << msg;
    int option;

    while(true){
        std::cout << input_msg;
        std::cin >> option;

        if(option < start || option > end || std::cin.peek() != '\n'){
            std::cout << "Invalid option! Choose between " << start << " to " << end << "\n";
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            continue;
        }
        
        return option;
    }
}

float input_x(std::string msg="Enter value: "){
    float value;

    while(true){
        std::cout << msg;
        std::cin >> value;

        if(std::cin.fail() || std::cin.peek() != '\n'){
            std::cout << "Invalid input! Value must be a number.\n";
            std::cin.clear();
            std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
            continue;
        }
        
        return value;
    }
}

